### PR TITLE
Added support for URLScan and PassiveTotal

### PR DESCRIPTION
**Pull request by caffix**
#71

### CODE REVIEW

File: amass/shodan.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ type Shodan struct {
 	BaseService
 
 	SourceType string
-	APIKey     *APIKey
+	API        *APIKey
```
Comment (fork-while-fork) [inline amass/shodan.go]: I think we should standardize how API credentials should be stored/scoped. Looks like most of the sources  store the credentials in a function local var while this one stores it as a member of the service struct.

File: amass/binaryedge.go [hunk 2]
```diff
@@ -62,21 +62,18 @@ func (be *BinaryEdge) startRootDomains() {
 }
 
 func (be *BinaryEdge) executeQuery(domain string) {
-	var err error
-	var url, page string
-
-	key := be.Enum().Config.GetAPIKey(be.String())
-	if key == nil {
+	api := be.Enum().Config.GetAPIKey(be.String())
+	if api == nil {
```
Comment (fork-while-fork) [inline amass/binaryedge.go]: I can see forgetting to set an API key being a common error. We should have log messages when the API key isn't supplied.

Comment (caffix) [reply amass/binaryedge.go]: All the data sources that use API keys now log a failure to acquire the correct values

**Comment by caffix** [comment]
Support has been added for CIRCL passive DNS as a data source

**Pull request merged**